api/server/fntracer.go

```diff
@@ -0,0 +1,130 @@
+type FnSpan struct {
+	wrappedSpan opentracing.Span
+}
+
+// NewFnSpan returns a new FnSpan which wraps the specified Span
+func NewFnSpan(spanToWrap opentracing.Span) opentracing.Span {
+	newSpan := &FnSpan{}
+	newSpan.wrappedSpan = spanToWrap
+	return newSpan
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) Finish() {
+	thisFnSpan.copyBaggageItemsToTags()
+	thisFnSpan.wrappedSpan.Finish()
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) FinishWithOptions(opts opentracing.FinishOptions) {
+	thisFnSpan.copyBaggageItemsToTags()
+	thisFnSpan.wrappedSpan.FinishWithOptions(opts)
+}
+
+func (thisFnSpan FnSpan) copyBaggageItemsToTags() {
+	// copy baggage items (which are inherited from the parent) with keys starting with "fn" to tags
+	// the PrometheusCollector will send these to Prometheus
+	// need to do this because the collector can't access baggage items, but it can access tags
+	// whereas here we can access the parent's baggage items, but not its tags
+	thisFnSpan.Context().ForeachBaggageItem(func(k, v string) bool {
+		if strings.HasPrefix(k, "fn") {
+			thisFnSpan.SetTag(k, v)
+		}
+		return true
+	})
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) Context() opentracing.SpanContext {
+	return thisFnSpan.wrappedSpan.Context()
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) SetOperationName(operationName string) opentracing.Span {
+	return thisFnSpan.wrappedSpan.SetOperationName(operationName)
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) SetTag(key string, value interface{}) opentracing.Span {
+	return thisFnSpan.wrappedSpan.SetTag(key, value)
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) LogFields(fields ...log.Field) {
+	thisFnSpan.wrappedSpan.LogFields(fields...)
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) LogKV(alternatingKeyValues ...interface{}) {
+	thisFnSpan.wrappedSpan.LogKV(alternatingKeyValues...)
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) SetBaggageItem(restrictedKey, value string) opentracing.Span {
+	return thisFnSpan.wrappedSpan.SetBaggageItem(restrictedKey, value)
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) BaggageItem(restrictedKey string) string {
+	return thisFnSpan.wrappedSpan.BaggageItem(restrictedKey)
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) Tracer() opentracing.Tracer {
+	return thisFnSpan.wrappedSpan.Tracer()
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) LogEvent(event string) {
+	thisFnSpan.wrappedSpan.LogEvent(event)
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) LogEventWithPayload(event string, payload interface{}) {
+	thisFnSpan.wrappedSpan.LogEventWithPayload(event, payload)
+}
+
+// FnSpan implements opentracing.Span
+func (thisFnSpan FnSpan) Log(data opentracing.LogData) {
+	thisFnSpan.wrappedSpan.Log(data)
+}
```

Comment:
Agree with @rdallman, most of this methods would go away when you'd embed  `opentracing.Tracer`.

Reply:
Yes, agreed. That's a very nice feature: no boilerplate proxying.